server/src/main/java/org/elasticsearch/index/query/TermsSetQueryBuilder.java

```diff
@@ -342,6 +353,108 @@ public LongValuesSource rewrite(IndexSearcher searcher) throws IOException {
 
     }
 
+    private static final Set<Class<? extends Number>> INTEGER_TYPES = new HashSet<>(
```

Comment:
#21776 made an effort to convert the List<?> in some special cases to help GC. But if we just serialize the list to a BytesRef, the convertion can be unnecessary. So i moved the method to the TermsSetQueryBuilder, which is the other query needs this method. I think TermsSetQueryBuilder's values do not need to be serialized because its terms number is limited to 1024.

Reply:
I agree, we can consider this PR as another approach to handle large `terms` query but this is not needed for the `terms_set` query. Maybe that's a chance to cleanup this code a bit and remove the optimization entirely here if it's not needed. That can also be done in a follow up but in such case please add a TODO regarding the removal of this code.

Reply:
I'd like to fix this in a folow up. A TODO note has been added here.

Reply:
++, please use `TODO` in capital letters for consistency.

Reply:
Thanks! fixed in [b2b9afd](https://github.com/elastic/elasticsearch/pull/67223/commits/b2b9afd5e81a9921abef3cbf25aff1d9083d87b0).